Api/Modules/Queries/Services/StyledOutputService.cs

```
@@ -216,24 +209,34 @@ private async Task<ServiceResult<string>> GetStyledOutputResultAsync(ClaimsIdent
                     ErrorMessage = errorMsg
                 };             
             }
+            
+            try
+            {
+                if (!QueryIsAllowed(style.QueryId, identity).Result)
+                {
+                    var errorMsg = $"Wiser user '{IdentityHelpers.GetUserName(identity)}' has no permission to execute query '{style.QueryId}'";
+                    logger.LogError(errorMsg);
 
-            if ((await wiserItemsService.GetUserQueryPermissionsAsync(formatQueryId, IdentityHelpers.GetWiserUserId(identity)) & AccessRights.Read) == AccessRights.Nothing)
+                    return new ServiceResult<string>
+                    {
+                        StatusCode = HttpStatusCode.Unauthorized,
+                        ErrorMessage = errorMsg
+                    };
+                }
+            }
+            catch (Exception e)
             {
-                var errorMsg = $"Wiser user '{IdentityHelpers.GetUserName(identity)}' has no permission to execute query '{formatQueryId}' for styled output '{id}' .";
-                
-                logger.LogError(errorMsg);
-                
                 return new ServiceResult<string>
                 {
                     StatusCode = HttpStatusCode.Unauthorized,
-                    ErrorMessage = errorMsg
+                    ErrorMessage = e.Message
```

Same here, please use `ILogger` to log the entire exception, to make debugging easier.

---

there was already a logger in the calling parent function but i guess never hurts to have more logging, added :)

---

This is not what I meant. Now you're still only logging `e.Message`. Please log the _entire_ exception, not just the Message. Your log should be something like this:
```C#
logger.LogError(e, "An error occurred while checking if a query is allowed to be executed.");
```
The first parameter is always for logging exceptions, the second is to add a message with more information.

---

was unaware that the logger also eats exceptions, changes made per request :)